test/ADtests.jl

```diff
@@ -229,32 +229,36 @@ for consf in [cons, con2_c]
     @test sol1.u ≈ sol2.u
 
     optf1 = OptimizationFunction(rosenbrock, Optimization.AutoFiniteDiff(); cons = consf)
-    lcons = consf == cons ? [0.2] : [0.2, 0.33]
-    ucons = consf == cons ? [0.55] : [0.55, 0.81]
+    lcons = consf == cons ? [0.2] : [0.2, -0.81]
+    ucons = consf == cons ? [0.55] : [0.55, -0.1]
     prob1 = OptimizationProblem(optf1, [0.3, 0.5], lb = [0.2, 0.4], ub = [0.6, 0.8], lcons = lcons, ucons = ucons)
-    sol1 = solve(prob1,IPNewton())
+    sol1 = solve(prob1,Optim.SAMIN(), maxiters = 10000) # a lot of iterations... doesn't even converge actually
     optf2 = OptimizationFunction(rosenbrock, Optimization.AutoForwardDiff(); cons = consf)
     prob2 = OptimizationProblem(optf2, [0.3, 0.5], lb = [0.2, 0.4], ub = [0.6, 0.8], lcons = lcons, ucons = ucons)
-    sol2 = solve(prob2,IPNewton())
-    @test sol1.minimum ≈ sol2.minimum 
+    sol2 = solve(prob2,Optim.SAMIN(), maxiters = 10000)
```

Comment:
`SAMIN` doesn't use the constraints though?

Reply:
But it doesn't error if constraints are given? Optim just lets them through?

Reply:
I thought it did? Based on what is givne here http://optimization.sciml.ai/stable/optimization_packages/optim/#With-Constraint-Equations

Reply:
I did miss that SAMIN is derivative-free though (PSO too), so we should probably mark that in the docs.